### PR TITLE
[feature-wip](unique-key-merge-on-write) fix that version is awlays 0 when update delete bitmap

### PR DESCRIPTION
**Pull request by liaoxin01**
# Proposed changes

Issue Number: close #xxx

## Problem summary
introduced by pr #11953.
Version has not been set when update delete bitmap. 

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by gavinchou** [comment]
Nice catch!

File: be/src/olap/txn_manager.cpp [hunk 1]
```diff
@@ -309,6 +309,9 @@ Status TxnManager::publish_txn(OlapMeta* meta, TPartitionId partition_id,
         // save meta need access disk, it maybe very slow, so that it is not in global txn lock
         // it is under a single txn lock
         if (rowset_ptr != nullptr) {
+            // TODO(ygl): rowset is already set version here, memory is changed, if save failed
```
Comment (zhannngchen) [inline be/src/olap/txn_manager.cpp:312]: The comments should be removed? L336 returned an OLAPInternalError

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**